### ISSUE TITLE
Servicemonitor CRD schema allows invalid input

### ISSUE THREAD

**Issue by janotav** [prometheus-operator/prometheus-operator#3942]
**What happened?**

User uploaded invalid servicemonitor CR that was using invalid value in metricRelabeling action field. Prometheus-operator detects the configuration change but Prometheus fails to reload the configuration. If this happens during start up, Prometheus won't start at all.

**Did you expect to see something different?**

I expect the invalid servicemonitor to be rejected upon creation by CRD schema validation. Currently any string is accepted. Only the predefined set recognized by Prometheus should be allowed.

**How to reproduce it (as minimally and precisely as possible)**:

```
spec:
  endpoints:
  - metricRelabelings:
    - action: test
```
leads to 

```
level=error ts=2021-03-28T14:31:50.344Z caller=main.go:656 msg="Error reloading config" err="couldn't load configuration (--config.file=\"/etc/prometheus/config_out/prometheus.env.yaml\"): parsing YAML file /etc/prometheus/config_out/prometheus.env.yaml: unknown relabel action \"test\""
```

**Environment**

* Prometheus Operator version: 0.39.0

* Kubernetes version information:

```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2", GitCommit:"52c56ce7a8272c798dbc29846288d7cd9fbae032", GitTreeState:"clean", BuildDate:"2020-04-16T11:56:40Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.7", GitCommit:"6c143d35bb11d74970e7bc0b6c45b6bfdffc0bd4", GitTreeState:"clean", BuildDate:"2019-12-11T12:34:17Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind:

    insert how you created your cluster: kops, bootkube, etc.

* Manifests:

```
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  labels:
    app: myapp
  name: myapp
spec:
  endpoints:
  - metricRelabelings:
    - action: test
    path: /metrics
    port: http
  namespaceSelector:
    matchNames:
    - default
  selector:
    matchLabels:
      app: myapp

```

* Prometheus Operator Logs: N/A

**Anything else we need to know?**:

**Comment by simonpasquier** [comment]
Thanks for the report! Do you want to give a try? 

Somehow related to https://github.com/prometheus-operator/prometheus-operator/pull/3914 but this particular issue needs to be fixed anyway.

**Comment by mikechengwei** [comment]
I will fix this issus.

**Comment by slashpai** [comment]
This needs to reopen :)

**Comment by simonpasquier** [comment]
closed by https://github.com/prometheus-operator/prometheus-operator/pull/4429

**Comment by mkoertgen** [comment]
The fix of #4429 seems to break existing labelMap-actions. With v0.54 using kube-stack we are lossing some PodMonitors (in this case from our Pulsar Cluster.

Log from the operator
```
level=warn ts=2022-02-10T21:36:06.190554556Z caller=operator.go:1957 component=prometheusoperator msg="skipping podmonitor" error="\"\" is invalid 'replacement' for labelmap action" podmonitor=pulsar/pulsar-zookeeper namespace=observability prometheus=prometheus-community-kube-prometheus
level=warn ts=2022-02-10T21:36:06.190682758Z caller=operator.go:1957 component=prometheusoperator msg="skipping podmonitor" error="\"\" is invalid 'replacement' for labelmap action" podmonitor=pulsar/pulsar-broker namespace=observability prometheus=prometheus-community-kube-prometheus
```

From looking at the podMonitors
```
...
    relabelings:
        - action: labelmap
          regex: __meta_kubernetes_pod_label_(.+)
```

this should be the problematic action.

**Comment by mkoertgen** [comment]
I manually deleted the labelmap action and voila we have metrics again. 

However, when validating the targetlabel-match maybe empty labels would probably just be skipped by prometheus?
Note that according to the Docs the labelMap-action does not require a target_label:
- https://prometheus.io/docs/prometheus/latest/configuration/configuration/

**Comment by simonpasquier** [comment]
@mkoertgen thanks for the report I've create an issue (#4572) to track it as I believe that you're right.

**Comment by slashpai** [comment]
Thanks for the reporting the bug. I will take a look at it :)

**Comment by slashpai** [comment]
The fix is available in 0.54.1
